snapcraft/internal/pluginhandler.py

```diff
@@ -250,8 +250,14 @@ def _step_state_file(self, step):
         return os.path.join(self.statedir, step)
 
     def _fetch_stage_packages(self):
-        if self.code.stage_packages:
+        if not self.code.stage_packages:
+            return
+
+        try:
             self.ubuntu.get(self.code.stage_packages)
+        except repo.PackageNotFoundError as e:
+            raise RuntimeError("Cannot find {!r} defined as 'stage-packages' "
+                               "for {!r}".format(e.package_name, self.name))
```

Comment:
"Cannot find 'libfoo' defined as 'stage-packages' for 'bar'" doesn't read very well. What about "Error downloading stage packages for part 'bar': no such package 'libfoo'" ?